File: r/cartclass.r
# Import dataset
df = read.csv('data.csv')

# Create Boolean vector with random split
library(caTools)
split = sample.split(df$y, SplitRatio=0.8)

# Split into training and test set
train = subset(df, split == TRUE)
test = subset(df, split == FALSE)

# Recursive Partitioning (RPART) for classification, regression,
# and survival trees. Implements most functionality from 1984
# book by Breiman, Friedman, Olshen and Stone. 
install.packages('rpart')
library(rpart)

# Fit CART classifier to training set
model = rpart(y ~ ., data=train, 
							control=rpart.control(minsplit=1))

# Predict test set
y_pred = predict(model, newdata=test[, -ncol(test)], type="class")

# Plot dendrogram
plot(model)
text(model)

File: python/cartclass.py
import numpy as np
import pandas as pd

# Import dataset
df = pdf.read_csv('data.csv')
X = df.iloc[:, :-1].values
y = df.iloc[:, -1].values

# Split into training and test sets
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

# Fit CART classifier to training set
from sklearn.tree import DecisionTreeClassifier
model = DecisionTreeClassifier()
model.fit(X_train, y_train)

# Predict test set
y_pred = model.predict(X_test)